Set `display` to `column_id` if not defined.

esoergel opened the pull request:
from https://github.com/dimagi/commcare-hq/pull/8781
@czue @benrudolph

esoergel committed:
Set `display` to `column_id` if not defined.

czue commented on corehq/apps/userreports/reports/specs.py:
```diff
@@ -65,6 +65,12 @@ class ReportColumn(JsonObject):
+    @classmethod
+    def wrap(cls, obj):
+        if 'display' not in obj:
+            obj['display'] = obj['column_id']
```
do you need to check if `column_id` is in `obj`? I think if it's not that might raise a `KeyError` which then might not be caught and raised by the framework as a `BadSpecError` (which is desired since those are caught and shown to the user).

would also be nice to add test coverage for this. for example, [like this](https://github.com/dimagi/commcare-hq/pull/8814/files#diff-65c4a40cc04d6a20143b4b51b3a1f86dR88)

esoergel replied:
Sounds good, will update.

esoergel committed:
Test fallback to column_id

esoergel committed:
Failing test that missing `column_id` raises `BadSpecError`

This one should be failing a lot more often - we still have a "lazy
migration" which is obscuring failures and not making this required.

esoergel committed:
Catch JsonObject errors and raise as BadSpecErrors

A missing `column id` would fail with an uncaught `BadValueError`

esoergel committed:
`pass` on missing `column_id`, so it's caught by JsonObject

esoergel commented:
@czue @nickpell This PR now does a few things, commit-by-commit is probably easiest.

czue commented:
looks good, though test fail looks real:
```
======================================================================
ERROR: testBadAggregation (corehq.apps.userreports.tests.test_columns.TestFieldColumn)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/tests/test_columns.py", line 55, in testBadAggregation
    "type": "field",
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/reports/factory.py", line 131, in from_spec
    str(e),
BadSpecError: Problem creating column from spec: {
  "field": "doc_id", 
  "display": "doc_id", 
  "type": "field", 
  "aggregation": "simple_", 
  "column_id": "doc_id"
}, message is: u'simple_' not in choices: ['count', 'count_unique', 'min', 'simple', 'max', 'sum', 'avg', 'year', 'month']
```

esoergel committed:
Update tests to check for BadSpecError

esoergel commented:
Tests passing.  I'll drop the lazy migration issue for now.
